torchbox/wagtail-torchbox
# Add Work and Thinking to the main proposition pages

[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1196138420)

### Description of Changes Made

This MR allows for addition of work and thinking blocks to the PropositionPage. 

The PropositionPage has three sections: services, clients, team. Each section has a body streamfield, which consists of the blocks defined in `tbx.core.blocks.PageSectionStoryBlock`.

Instead of modifying `PageSectionStoryBlock`, I have subclassed it and created a new `tbx.propositions.blocks.PropositionPageStoryBlock`, which includes `work` & `thinking` blocks. This approach ensures that we do not affect other Page models that use the `PageSectionStoryBlock`. Further, if we need to make other changes to the PropositionPage's StreamFields in the future, we can just focus on the `PropositionPageStoryBlock`.

> **Note**
>
> this needs some FE work to modify the templates. Two possible approaches:
>
> - copy the templates for the work and thinking blocks, modify them and pass the new templates when instantiating the blocks, for instance `work = WorkBlock(template='path/to/new/template.html')`
> - modify the existing templates for the work and thinking blocks, so that appropriate CSS classes can be passed to the templates. The template will render differently depending on the CSS classes that load.

### How to Test

- When editing an existing Proposition Page, the body streamfield for the services, clients & team sections should now have *work* and *thinking* blocks.
- When you add either a work or thinking block under any section, you should see it on the frontend. However, at this point, the styling will not make sense (please see screenshots below), because these blocks were designed to be individual, standalone sections, just like the services, clients & team sections.

### Screenshots

<details>
  <summary>Expand to see more</summary>

![image](https://github.com/torchbox/wagtail-torchbox/assets/7713776/7788c2c2-0b56-4180-9515-21a0c509ed53)

![image](https://github.com/torchbox/wagtail-torchbox/assets/7713776/ef10c589-be4d-4c83-be4f-22ab92a93d77)

</details>

### MR Checklist
